rig-core/src/vector_store/in_memory_store.rs

```diff
@@ -7,28 +7,29 @@ use std::{
 use ordered_float::OrderedFloat;
 use serde::{Deserialize, Serialize};
 
-use super::{VectorStore, VectorStoreError, VectorStoreIndex};
-use crate::embeddings::{DocumentEmbeddings, Embedding, EmbeddingModel, EmbeddingsBuilder};
+use super::{VectorStoreError, VectorStoreIndex};
+use crate::embeddings::{Embedding, EmbeddingModel};
 
 /// InMemoryVectorStore is a simple in-memory vector store that stores embeddings
 /// in-memory using a HashMap.
-#[derive(Clone, Default, Deserialize, Serialize)]
-pub struct InMemoryVectorStore {
-    /// The embeddings are stored in a HashMap with the document ID as the key.
-    embeddings: HashMap<String, DocumentEmbeddings>,
+#[derive(Clone, Default)]
+pub struct InMemoryVectorStore<D: Serialize> {
+    /// The embeddings are stored in a HashMap.
+    /// Hashmap key is the document id.
+    /// Hashmap value is a tuple of the serializable document and its corresponding embeddings.
+    embeddings: HashMap<String, (D, Vec<Embedding>)>,
 }
 
-impl InMemoryVectorStore {
+impl<D: Serialize + Eq> InMemoryVectorStore<D> {
     /// Implement vector search on InMemoryVectorStore.
     /// To be used by implementations of top_n and top_n_ids methods on VectorStoreIndex trait for InMemoryVectorStore.
-    fn vector_search(&self, prompt_embedding: &Embedding, n: usize) -> EmbeddingRanking {
+    fn vector_search(&self, prompt_embedding: &Embedding, n: usize) -> EmbeddingRanking<D> {
```

Comment:
Add unit tests for `vector_search` (at least one with multiple embeddings per item). 

Populate the store with fake embeddings with very few dimensions but make sure that are sensible. I.e.: Make the matching vectors close to each other (e.g.: `[0.1, 0.1, 0.5]` and `[0.1, 0.1, 0.6]`) and non-matching vectors away from each other (e.g.: `[0.1, 0.1, 0.5]` and `[-1.0, -0.6, 0.1]`)